lmfit/model.py

```diff
@@ -1116,9 +1116,9 @@ def __init__(self, left, right, op, **kws):
                         "use distinct names.")
             raise NameError(msg)
 
-        # we assume that all the sub-models have the same independent vars
+        # every independent_vars of the left and right model is added to independent_vars of the CompositeModel
         if 'independent_vars' not in kws:
-            kws['independent_vars'] = self.left.independent_vars
+            kws['independent_vars'] = list(dict.fromkeys([*self.left.independent_vars, *self.right.independent_vars]))
```

Comment:
The functional change looks fine I think. But first making a dictionary and then  converting it to a list seems a bit awkward to me. Is this the easiest/cleanest way of doing it (I haven't checked myself yet)?

Reply:
perhaps 
 
```
       ivar1, ivar2 =  self.left.independent_vars,  self.right.independent_vars
       kws['independent_vars'] = ivar1 + [x for x in ivar2 if x not in ivar1] 
```

Reply:
I have reviewed my solution and the alternative suggestions by @newville. Performance-wise the solution of making a dictionary and later converting it back to a list is generally faster. If the lengths of lists of independent_vars are in the range of n=100 by a factor of 30, for shorter and more reasonable lengths of n=10 it is only a factor of 2. 
Obviously, in typical cases, the execution speed will never play a role because we will probably never have large numbers of independent parameters. I will therefore use a version similar to @newville 's one, for better readability.